### PR TITLE
reduce the number of `IMessageExtractor` invocations

### PR DESCRIPTION
**Pull request by Aaronontheweb**

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Part of #5203

**Review by Arkatufus — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
Looks like we have a legit MNTR failure - taking a look at that

**Comment by Aaronontheweb** [comment]
I'm going to have to redo parts of this

File: src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs [hunk 1]
```diff
@@ -658,12 +658,15 @@ bool AwaitingLease(object message)
...
                     return true;
-
-                case var _ when shard.ExtractEntityId(message).HasValue:
-                    shard.DeliverMessage(message, shard.Context.Sender);
+                default:
+                {
+                    var extracted = shard.ExtractEntityId(message);
+                    if (!extracted.HasValue) return false;
+                    
+                    shard.DeliverTo(extracted.Value.Item1, message, extracted.Value.Item2, shard.Context.Sender);
```
Comment (to11mtm) [inline src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:666]: Is there a reason this got changed to `shard.DeliverTo` when the old code did `shard.DeliverMessage`?

**Comment by Aaronontheweb** [comment]
This was superseded by https://github.com/akkadotnet/akka.net/pull/6863

**Pull request closed without merging**